src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs

```diff
@@ -144,21 +144,15 @@ internal class RPackageManager : IRPackageManager {
             }
         }
 
-        public PackageLockState GetPackageLockState(string name, string libraryPath) {
-            string dllPath = GetPackageDllPath(name, libraryPath);
-            if (!string.IsNullOrEmpty(dllPath)) {
-                var processes = RestartManager.GetProcessesUsingFiles(new string[] { dllPath }).ToArray();
-                if (processes != null) {
-                    if (processes.Length == 1 && processes[0].Id == _interactiveWorkflow.RSession.ProcessId) {
-                        return PackageLockState.LockedByRSession;
-                    }
+        public async Task<PackageLockState> GetPackageLockState(string name, string libraryPath) {
+            var pkgStateresult =  await _interactiveWorkflow.RSession.EvaluateAsync($"rtvs:::package_lock_state({name.ToRStringLiteral()}, {libraryPath.ToRStringLiteral()})", REvaluationKind.Normal);
```

Comment:
Use `EvaluateAsync<string>()` extension method to get a typed value back directly.

Better yet, you can use [this](http://www.newtonsoft.com/json/help/html/t_newtonsoft_json_converters_stringenumconverter.htm), and then do `EvaluateAsync<PackageLockState>()`. If you change `rtvs:::package_lock_state` to return a factor instead of a string, then you won't even need the attribute, since enum-as-int is default serialization behavior.